etc/errors/anvil.json

```diff
@@ -780,6 +780,321 @@
               }
             }
           ]
+        },
+        {
+          "component_name": "AnvilNode",
+          "component_code": 10,
+          "identifier_encoding": "node",
+          "errors" : [
+            {
+              "name": "TransactionValidationFailedGasLimit",
+              "code": 1,
+              "message": "Gas limit for transaction {transaction_hash} is {tx_gas_limit} which exceeds maximum allowed gas {max_gas}",
+              "fields": [
+                {
+                  "name": "transaction_hash",
+                  "type": "H256"
+                },
+                {
+                  "name": "tx_gas_limit",
+                  "type": "U256"
+                },
+                {
+                  "name": "max_gas",
+                  "type": "U256"
+                }
+              ],
+              "doc": {
+                "summary": "Transaction validation failed due to excessive gas limit.",
+                "description": [
+                  "This error occurs when a transaction's gas limit exceeds the maximum allowed gas.",
+                  "Anvil-zksync enforces gas limits that simulate those of the real ZKSync network to ensure testing accuracy.",
+                  "Note: When anvil-zksync is in forking mode, it lock-in and uses gas price at the forked block"
```

Comment:
Second line, although true, is irrelevant here. Please remove.

First one is IMHO a bit confusing. The limit, I am pretty sure, just comes from protocol limitations. I would rephrase accordingly something along the lines of: "Maximum allowed gas is regulated by ZKsync protocol and, as of protocol version 27, is limited to 2^50".

Realistically trusted gas limit is much lower than that set at 80M gas but zksync-era doesn't validate that (probably because of bootloader overhead that is hard to account for).

IMHO the only realistic scenario when this error would trigger is that user accidentally set insane gas limit for transaction by mistake (e.g. `40 * 10^60` instead of `40 * 10^6` gas etc). We should emphasize this in the error message